src/phoenix/server/api/dataloaders/cache/two_tier_cache.py

```diff
@@ -0,0 +1,72 @@
+from abc import ABC
+from asyncio import Future
+from typing import Any, Callable, Generic, Optional, Tuple, TypeVar, cast
+
+from cachetools import Cache, LFUCache
+from strawberry.dataloader import AbstractCache
+
+_Key = TypeVar("_Key")
+_Result = TypeVar("_Result")
+
+_Section = TypeVar("_Section")
+_SubKey = TypeVar("_SubKey")
+
+
+class TwoTierCache(
+    AbstractCache[_Key, _Result],
+    Generic[_Key, _Result, _Section, _SubKey],
+    ABC,
+):
+    _main_cache_factory: "staticmethod[[], Cache[_Section, Cache[_SubKey, Future[_Result]]]]" = (
+        staticmethod(lambda: LFUCache(maxsize=32))
+    )
+    _sub_cache_factory: "staticmethod[[], Cache[_SubKey, Future[_Result]]]" = staticmethod(
+        lambda: LFUCache(maxsize=8)
+    )
+
+    def __init_subclass__(
+        cls,
+        *,
+        main_cache_factory: Optional[
+            Callable[[], "Cache[_Section, Cache[_SubKey, Future[_Result]]]"]
+        ] = None,
+        sub_cache_factory: Optional[Callable[[], "Cache[_SubKey, Future[_Result]]"]] = None,
```

Comment:
These strike me as details that should be encapsulated within the class rather than injected in this way.

Reply:
it is encapsulated in the class definition

Reply:
but you also want the ability to inject it if you want a different eviction policy that's other than LFU